backend/service/inviteContact.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable no-case-declarations */
+import { type User, type Contact } from '@prisma/client';
+import SesService from '../awsSes/SesService';
+import PlanInviteModel from '../models/planInvite';
+
+enum PlanInviteStatus {
+  HAS_INVITED,
+  NOT_INVITED,
+  INVITE_EXPIRED
+}
+
+export default class InviteContactService {
+  static async setupEmailInviteToContacts(planLeader: User, planID: string, contactsArray: Contact[], message: string): Promise<Contact[] | null> {
+    // TODO: make contact email required in the contacts model
+    try {
+      const alreadyInvitedContacts: Contact[] = []
+      await Promise.all(contactsArray.map(async (contact: Contact) => {
+        const contactEmail = contact.email as string;
+
+        switch (await InviteContactService.checkPlanInviteStatus(planID, contactEmail)) {
+          case PlanInviteStatus.NOT_INVITED:
+            const planInviteResponse = await PlanInviteModel.dbCreateOnePlanInvite(planID, contactEmail);
+
+            if (planInviteResponse === null) throw new Error('planInviteResponse is null')
+
+            const url = InviteContactService.generateInviteLink(planInviteResponse.id);
+
+            await InviteContactService.sendEmailInviteToContact(planLeader, url, contactEmail, message)
+
+            break;
+          case PlanInviteStatus.INVITE_EXPIRED:
+            // resend invite
+            const expiredPlanInviteResponse = await PlanInviteModel.readOnePlanInvite(planID, contactEmail);
+
+            if (expiredPlanInviteResponse === null) throw new Error('expiredPlanInviteResponse is null')
+
+            const updatePlanInviteRecord = await PlanInviteModel.updatePlanInviteExpiration(expiredPlanInviteResponse.id)
+
+            if (updatePlanInviteRecord === null) throw new Error('modelResponse is null')
+
+            const newUrl = InviteContactService.generateInviteLink(updatePlanInviteRecord.id);
+
+            await InviteContactService.sendEmailInviteToContact(planLeader, newUrl, contactEmail, message)
+
+            break;
+          case PlanInviteStatus.HAS_INVITED:
+
+            alreadyInvitedContacts.push(contact)
+            break;
+        }
+      }))
```

Comment:
the way this block is structure implies something different.

You're calling `Promise.all` so we'd assume you'd end up with an Array of Promises. You're using a `map` which implies that you're convert one array to converting it to another. 

but as you'll notice in your block -- youre not actually returning any promises, in fact you're returning undefined in all cases, so you have a `Promise<undefined>[]` return type right now. 

Also related, using HOF ,like Map andForEach, with Promises has some funky behavior. https://stackoverflow.com/questions/64978604/async-await-map-not-awaiting-async-function-to-complete-inside-map-function-befo#:~:text=map()%20is%20not%20async,not%20waiting%20for%20that%20promise.

TLDR: maps are not promise aware, so it can lead to some un-expected results. I don't think you do in this case because of the Promise.all, but just wanted to share.

Also I'm just kind of confused what this function does. why does this function return the list of "emails already invited" rather than the list of all emails invited?

Reply:
Also, in general, using a `push` inside of a `.map` usually implies that you're doing something a little smelly.